docs/terraformoptions.adoc

```diff
@@ -574,6 +574,33 @@ EOT
     }
   }
 }`
+
+|defined_tags
+|Tagging using administrator defined tags. This is specified in the form of a map of maps.
+| e.g.
+`defined_tags = {
+  vcn = {
+    environment = "dev"
```

Comment:
Defined tags need to include the tag namespace on the left in between quotes e.g.

```
    "cn.vcn" = "dev"
```

when "cn" is the tag namespace and "vcn" is the tag key whereas "dev" is the value. Note also that a defined tag can also have a list of values.

Tag namespaces and tag keys must already exist in order for the resources to be successfully created. Additionally, if the tag key is of type list, then the specified value must be a valid value.

As such, I think the default values should be empty as tags would typically be created by administrators.

Can you please:
1. update the example values so that the tag keys are between quotes ""
2. add documentation how to use the tags
3. add error handling in case the tag namespace, keys or values use are invalid

Reply:
Regarding the error handling, should I use preconditions? If I remember correctly, using them means that the Terraform minimum version will be 1.2 and not 1.0.0

Reply:
I've done another round of test and I now understand more where you're going with the node tags and you're in the right direction.

So, similar to _nodepool_ tags, we need to have common and tags for the nodes that are specific to the nodepool. So, https://github.com/valireds/terraform-oci-oke/blob/issue-560/modules/oke/nodepools.tf#L56 and https://github.com/valireds/terraform-oci-oke/blob/issue-560/modules/oke/nodepools.tf#L57 should read similar as https://github.com/valireds/terraform-oci-oke/blob/issue-560/modules/oke/nodepools.tf#L13 and https://github.com/valireds/terraform-oci-oke/blob/issue-560/modules/oke/nodepools.tf#L14 respectively except that you'll be extracting the `node` key instead of `node_pool`.

Re: using Terraform 1.2, I'm ok with bumping the minimum Terraform requirement if this means we have a more solid version. We can cut a new minor release to indicate our support for it. Are there any impact on state? I haven't seen any.